frontend/src/composables/use-sensitive-media.ts

```diff
@@ -0,0 +1,74 @@
+import { computed } from "vue"
+
+import type { Media } from "~/types/media"
+
+import { useUiStore } from "~/stores/ui"
+
+type SensitiveVisibilityState =
+  | "non-sensitive"
+  | "sensitive-shown"
+  | "sensitive-hidden"
```

Comment:
Nit: these could be individually exported from a constants file as they are being used for `==` matching below.

Reply:
Moved to the content-safety constants file in c69e77829; great suggestion!